chain/chain/src/test_utils.rs

```diff
@@ -1400,13 +1404,84 @@ impl ChainGenesis {
     }
 }
 
+/// A wrapper function around process_block that doesn't trigger all the callbacks
+/// Only used in tests
+pub fn process_block_test(
+    chain: &mut Chain,
+    me: &Option<AccountId>,
+    block: Block,
+) -> Result<Option<Tip>, Error> {
+    chain.process_block(
+        me,
+        MaybeValidated::from(block),
+        Provenance::PRODUCED,
+        |_| {},
+        |_| {},
+        |_| {},
+    )
+}
+
+use rand_chacha::rand_core::SeedableRng;
+use rand_chacha::ChaCha8Rng;
+use rand_distr::{Distribution, Exp};
+
+pub fn fill_mock_clock_utc(
+    mock_date: chrono::DateTime<Utc>,
+    num_samples: u64,
+    average_interval_in_miliseconds: f64,
+    seed: u64,
+) {
+    let exp = Exp::new(average_interval_in_miliseconds).unwrap();
+    let mut rng = ChaCha8Rng::seed_from_u64(seed);
+    let mut utc = mock_date.clone();
+    for _i in 0..num_samples {
+        let f = exp.sample(&mut rng) as i64;
+        let d = Duration::milliseconds(f);
+        if let Some(t) = utc.checked_add_signed(d) {
+            Clock::add_utc(utc.clone());
+            utc = t;
+        }
+    }
+}
+
+pub fn build_chain_from_random_data(data: &[u8]) {
+    if data.len() > 8 {
+        let (a, b) = data.split_at(8);
+        let seed = u64::from_be_bytes(a.try_into().unwrap());
+        let num_blocks = b.len() as u64;
+        let _mock_clock_guard = MockClockGuard::default();
+        fill_mock_clock_utc(
+            chrono::Utc.ymd(2020, 10, 1).and_hms_milli(0, 0, 3, 444),
+            num_blocks * 2 + 1,
+            100.0,
```

Comment:
this is the 'average interval' between blocks, right? I'd have expected it to be 1 second - or did you select 0.1 seconds on purpose?